gap/attr.gi

```diff
@@ -3431,3 +3431,42 @@ D -> DIGRAPHS_IsJoinSemilatticeAndJoinTable(D)[2]);
 InstallMethod(DigraphMeetTable, "for a digraph",
 [IsDigraph],
 D -> DIGRAPHS_IsMeetSemilatticeAndMeetTable(D)[2]);
+
+InstallMethod(DigraphDistanceMetrics, "for a digraph",
+[IsDigraph],
+function(G)
+  local ecc, c, u, v;
+  if not IsDigraph(G) then
+    Error("Input must be a digraph");
+  elif not IsStronglyConnectedDigraph(G) then
+    Error("Input digraph is not strongly connected; property undefined");
+  fi;
+  ecc := [];
+  for u in [1 .. DigraphNrVertices(G)] do
+    c := 0;
+    for v in [1 .. DigraphNrVertices(G)] do
+      if u <> v then
+        c := Maximum(c, DigraphShortestDistance(G, u, v));
+      fi;
+    od;
+    Add(ecc, c);
+  od;
+  return rec(
```

Comment:
I'm not 100% sure but I think the following is not particularly efficient because the computation of `Minimum(ecc)` and `Maximum(ecc)` will be repeated for each distinct value of `i` which is not such a good idea. Better to compute these values prior to this return statement.